crates/aranya-client/src/afc.rs

```diff
@@ -6,241 +6,155 @@
 #[derive(Clone, Debug)]
-pub(crate) enum State {
+enum State {
     /// A peer opened a connection with us.
     Accept(SocketAddr),
     /// We recieved an incoming message.
     Msg(SocketAddr),
 }
 
+/// An Aranya Fast Channel message.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Message {
+    /// The plaintext data.
+    pub data: Vec<u8>,
+    /// The address from which the message was received.
+    pub address: SocketAddr,
+    /// The channel from which the message was received.
+    pub channel: AfcId,
+    /// The Aranya Fast Channel label associated with the message.
+    pub label: Label,
+    /// The order of the message in the channel.
+    pub seq: Seq,
+}
+
+/// The size in bytes of `magic || len`.
+///
+/// See the wire format description.
+const WIRE_HEADER_SIZE: usize = 4 + 4;
+
+/// See the wire format description.
+const WIRE_MAGIC: &[u8; 4] = b"AFC\0";
+
+/// The maximum allowed size of a `StreamMsg`.
+///
+/// Helps prevent DoS attacks.
+// TODO(eric): make this configurable.
+const MAX_MSG_SIZE: u32 = 10 * 1024 * 1024;
+
 /// AFC messages.
 ///
-/// These messages are sent/received between AFC peers via the
-/// TCP transport.
+/// These messages are sent/received between AFC peers via the TCP transport.
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub(crate) enum Msg {
+enum StreamMsg {
     Ctrl(Ctrl),
     Data(Data),
 }
 
 /// An AFC control message.
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub(crate) struct Ctrl {
-    pub version: Version,
-    pub team_id: TeamId,
+struct Ctrl {
+    version: Version,
+    team_id: TeamId,
     /// Ephemeral command for AFC channel creation.
-    pub cmd: AfcCtrl,
+    cmd: AfcCtrl,
 }
 
 /// An AFC data (ciphertext) message.
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub(crate) struct Data {
+struct Data {
     version: Version,
     afc_id: AfcId,
     ciphertext: Vec<u8>,
 }
 
-/// The size in bytes of `magic || len`.
-///
-/// See the wire format description.
-const WIRE_HEADER_SIZE: usize = 4 + 4;
-
-/// See the wire format description.
-const WIRE_MAGIC: &[u8; 4] = b"AFC\0";
-
-/// The maximum allowed size of a [`Msg`].
-///
-/// Helps prevent DoS attacks.
-// TODO(eric): make this configurable.
-const MAX_MSG_SIZE: u32 = 10 * 1024 * 1024;
-
 /// Sends and receives AFC messages.
-pub(crate) struct Afc<S> {
+pub(crate) struct FastChannelsImpl<S> {
     /// The underlying AFC client.
     afc: Client<S>,
     /// Listens for incoming connections from peers.
     listener: TcpListener,
     /// Open TCP connections.
     // TODO(eric): prune unused/idle streams.
-    // TODO(eric): use different maps for streams we opened vs
-    // streams that peers opened.
+    // TODO(eric): use different maps for streams we opened vs streams that
+    // peers opened.
     streams: TcpStreams,
     /// All open channels.
-    chans: BTreeMap<AfcId, Chan>,
+    // TODO: expose FastChannel to the user and add more methods to it
```

Comment:
nit: conventionally we use `TODO(name)`. It helps b/c it's much quicker than stepping through git history to see who let the TODO (like when comments get moved around, etc.).